HW1/P4/P4.txt

```diff
@@ -1 +1,17 @@
-# Your discussion here
+1. Graph representation:
+My graph representation includes one entry per character, where each entry has the following format:
+(character name, [connected character #1, connected character #2, ...])
+i.e. the key is the character name, and the value is a list of all (directly) connected characters.
+
+When loading the data to create the list, I clean up the data by removing quotation marks around both the character name and the comic book name. I also strip out trailing whitespace and aggregate characters with otherwise identical names (e.g. 'QUASAR DOPPELGANGER ' vs 'QUASAR DOPPELGANGER'). This results in a lower overall number of unique characters, as well as a potentially lower number of nodes touched during the breadth-first search (depending on the given start node).
+
+2. Search results:
+6403 touched nodes when starting from CAPTAIN AMERICA
```

Comment:
Yeah, about the same answer, to within 5 nodes. Probably due to your cleaning of the initial data.